google_guest_agent/hostname/hostname_test.go

```diff
@@ -0,0 +1,537 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package hostname
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/cfg"
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/command"
+	mds "github.com/GoogleCloudPlatform/guest-agent/metadata"
+)
+
+func TestReconfigureHostname(t *testing.T) {
+	cfg.Load(nil)
+	setFqdnOrig := setFqdn
+	setHostnameOrig := setHostname
+	t.Cleanup(func() { setFqdn = setFqdnOrig; setHostname = setHostnameOrig })
+	testcases := []struct {
+		name         string
+		cfg          *cfg.Sections
+		lastHostname string
+		lastFqdn     string
+		setFqdn      func(string, string) error
+		setHostname  func(string) error
+		req          ReconfigureHostnameRequest
+		resp         ReconfigureHostnameResponse
+	}{
+		{
+			name: "successful reconfigure all",
```

Comment:
please avoid using space and special characters to test names.

Reply:
Done.